### PR TITLE
feat: unify richtext config

### PR DESCRIPTION
**Pull request by xiaoluoHe**
[[中文版模板 / Chinese template](https://github.com/VisActor/vrender/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Release
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vrender-components/src/tag/tag.ts [hunk 1]
```diff
@@ -166,7 +130,7 @@ export class Tag extends AbstractComponent<Required<TagAttributes>> {
       }
     } else {
       const textAttrs = {
-        text: text as string | number | string[] | number[],
+        text: isObject(text) && 'type' in text && text.type === 'text' ? text.text : text,
```
Comment (xile611) [inline packages/vrender-components/src/tag/tag.ts:133]: 这一段要不要页抽取一个公共方法？

File: packages/vrender-components/src/legend/discrete/discrete.ts [hunk 2]
```diff
@@ -417,31 +418,46 @@ export class DiscreteLegend extends LegendBase<DiscreteLegendAttrs> {
...
       y: 0,
       textAlign: 'start',
       textBaseline: 'middle',
       lineHeight: (labelStyle.style as ITextGraphicAttribute)?.fontSize,
       ...labelStyle.style,
-      text: labelAttr.formatMethod ? labelAttr.formatMethod(label, item, index) : label
-    });
+      text
+    };
+    if (isRichText({ text })) {
```
Comment (xile611) [inline packages/vrender-components/src/legend/discrete/discrete.ts:432]: 这里不需要创建新的对象吧，直接传入 `labelAttributes`就可以了吧

File: packages/vrender-components/src/label/base.ts [hunk 3]
```diff
@@ -295,25 +302,10 @@ export class LabelBase<T extends BaseLabelAttrs> extends AbstractComponent<T> {
...
-    } else if (attributes.textType === 'html') {
-      attributes.textConfig = [] as IRichTextCharacter[];
-      attributes.html = {
-        dom: attributes.text as string,
-        ...DEFAULT_HTML_TEXT_SPEC,
-        ...attributes
-      };
-      const text = graphicCreator.richtext(attributes as IRichTextGraphicAttribute);
-      return text;
+    if (isRichText(attributes as any, 'textType')) {
```
Comment (xile611) [inline packages/vrender-components/src/label/base.ts:305]: 现在只有 `label`继续支持 `textType`?

Comment (xiaoluoHe) [reply packages/vrender-components/src/label/base.ts:305]: 是的，label 的配置里有 type 字段了 😂

**Review by xile611 — APPROVED** [review]

**Pull request merged**